raft.go

```diff
@@ -1484,3 +1664,66 @@ func (r *Raft) setState(state RaftState) {
 		r.observe(state)
 	}
 }
+
+// LookupServer looks up a server by ServerID.
+func (r *Raft) lookupServer(id ServerID) *Server {
+	for _, server := range r.configurations.latest.Servers {
+		if server.ID != r.localID {
+			return &server
+		}
+	}
+	return nil
+}
+
+// pickServer returns the follower that is most up to date.
```

Comment:
Should we comment this is only to be called in leader loop?

Reply:
done!